owncloudApp/src/main/java/com/owncloud/android/presentation/accounts/ManageAccountsDialogFragment.kt

```diff
@@ -0,0 +1,267 @@
+
+        subscribeToViewModels()
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val builder = AlertDialog.Builder(ContextThemeWrapper(requireContext(), R.style.Theme_AppCompat_Dialog_Alert))
+        val inflater = this.layoutInflater
+        val dialogView = inflater.inflate(R.layout.manage_accounts_dialog, null)
+        builder.setView(dialogView)
+
+        val recyclerView = dialogView.findViewById<RecyclerView>(R.id.account_list_recycler_view)
+
+        recyclerView.apply {
+            filterTouchesWhenObscured = PreferenceUtils.shouldDisallowTouchesWithOtherVisibleWindows(requireContext())
+            adapter = accountListAdapter
+            layoutManager = LinearLayoutManager(requireContext())
+        }
+
+        val closeButton = dialogView.findViewById<ImageView>(R.id.cross)
+        closeButton.setOnClickListener {
+            dismiss()
+        }
+
+        val dialog = builder.create()
+        dialog.window?.setBackgroundDrawableResource(R.color.transparent)
+
+        return dialog
+    }
+
+    override fun removeAccount(account: Account) {
+        dismiss()
+        val hasAccountAttachedCameraUploads = manageAccountsViewModel.hasCameraUploadsAttached(account.name)
+        val dialog = AlertDialog.Builder(requireContext())
+            .setMessage(getString(
+                if (hasAccountAttachedCameraUploads) R.string.confirmation_remove_account_alert_camera_uploads
+                else R.string.confirmation_remove_account_alert, account.name)
+            )
+            .setPositiveButton(getString(R.string.common_yes)) { _, _ ->
+                val accountManager = AccountManager.get(MainApp.appContext)
+                accountManager.removeAccount(account, this, null)
+                if (manageAccountsViewModel.getLoggedAccounts().size > 1) {
+                    show(supportFragmentManager, MANAGE_ACCOUNTS_DIALOG)
+                }
+            }
+            .setNegativeButton(getString(R.string.common_no)) { _, _ ->
+                show(supportFragmentManager, MANAGE_ACCOUNTS_DIALOG)
+            }
+            .create()
+        dialog.avoidScreenshotsIfNeeded()
+        dialog.show()
+    }
+
+    override fun cleanAccountLocalStorage(account: Account) {
+        dismiss()
+        val dialog = AlertDialog.Builder(requireContext())
+            .setTitle(getString(R.string.clean_data_account_title))
+            .setIcon(R.drawable.ic_warning)
+            .setMessage(getString(R.string.clean_data_account_message, account.name))
+            .setPositiveButton(getString(R.string.clean_data_account_button_yes)) { _, _ ->
+                show(supportFragmentManager, MANAGE_ACCOUNTS_DIALOG)
+                manageAccountsViewModel.cleanAccountLocalStorage(account.name)
+            }
+            .setNegativeButton(R.string.drawer_close) { _, _ ->
+                show(supportFragmentManager, MANAGE_ACCOUNTS_DIALOG)
```

Comment:
If I'm not mistaken, when you press the `setNegativeButton` the manage accounts dialog should appear again, but what it does is close all dialogs and navigate you to the main list. Testing with the clean account dialog does work: it sends you to the manage accounts dialog.

Reply:
When you press "Close" in the clean account dialog, the "Manage accounts" dialog appears again, as it is specified in these lines. With the "No" button in the remove account dialog the same happens. I don't know what are you suggesting here